src/ape/utils.py

```diff
@@ -444,6 +454,45 @@ def get_all_files_in_directory(path: Path) -> List[Path]:
     return [path]
 
 
+class injected_before_use(property):
+    """
+    Injected properties are injected class variables that must be set before use
+    NOTE: do not appear in a Pydantic model's set of properties
+    """
+
+    def __get__(self, *args):
+        raise ValueError("Value not set. Please inject this property before calling.")
+
+
+class ManagerAccessBase:
```

Comment:
Maybe we should put this in `ape.managers.base.py` ?

Reply:
This would prevent it from being importable into the `ape.api` modules due to circular imports

Reply:
Ah, I see